code/modules/power/power_exporter.dm

```diff
@@ -0,0 +1,145 @@
+	..()
+	if(!active)
+		user << "<span class='notice'>The exporter seems to be offline.</span>"
+	else
+		user << "<span class='notice'>The [src] is exporting [drain_rate] kilowatts of power, it has consumed [power_drained] kilowatts so far.</span>"
+
+obj/machinery/power/exporter/Initialize()
+	..()
+	power_exporter_list += src
+
+obj/machinery/power/exporter/Destroy()
+	power_exporter_list -= src
+	return ..()
+
+/obj/machinery/power/exporter/attackby(obj/item/O, mob/user, params)
+	if(!active)
+		if(istype(O, /obj/item/weapon/wrench))
+			if(!anchored && !isinspace())
+				connect_to_network()
+				user << "<span class='notice'>You secure the [src] to the floor.</span>"
+				anchored = TRUE
+			else if(anchored)
+				disconnect_from_network()
+				user << "<span class='notice'>You unsecure and disconnect the [src].</span>"
+				anchored = FALSE
+			playsound(loc, 'sound/items/Deconstruct.ogg', 50, 1)
+			return
+	return ..()
+
+/obj/machinery/power/exporter/attack_hand(mob/user)
+	..()
+	if (!anchored)
+		user << "<span class='warning'>This device must be anchored by a wrench!</span>"
+		return
+	if (power_drained >= 10000000 && !rewarded) // 10 billion watts
+		rewarded = TRUE
+		visible_message("<span class='danger'>You have been blessed by the gods of engineering for your massive power exports!</span>")
+		new /obj/effect/overlay/temp/explosion/fast(get_turf(src))
+		playsound(src, 'sound/effects/pray_chaplain.ogg', 100, 1)
+		for(var/i in 1 to 3)
+			new /obj/item/clothing/head/orange_beret(get_turf(src))
+			new /obj/item/clothing/suit/poncho/engiponcho(get_turf(src))
+			new /obj/item/clothing/gloves/krav_maga/engi(get_turf(src))
+	interact(user)
+
+/obj/machinery/power/exporter/attack_ai(mob/user)
+	interact(user)
+
+/obj/machinery/power/exporter/attack_paw(mob/user)
+	interact(user)
+
+/obj/machinery/power/exporter/interact(mob/user)
+	if (!anchored)
+		user << "<span class='warning'>This device must be anchored by a wrench!</span>"
+		return
+	if(!Adjacent(user))
+		if(!isAI(user))
+			return
+	user.set_machine(src)
+	var/list/dat = list()
+	dat += ("<b>[name]</b><br>")
+	if (active)
+		dat += ("Exporter: <A href='?src=\ref[src];action=disable'>On</A><br>")
+	else
+		dat += ("Exporter: <A href='?src=\ref[src];action=enable'>Off</A><br>")
+	dat += ("Power consumption: <A href='?src=\ref[src];action=set_power'>[drain_rate] kilowatts</A></u></b><br><br>")
+	dat += ("<b><u>Surplus power: [(powernet == null ? "Unconnected" : "[powernet.netexcess/1000] kilowatts")]<br>")
+	dat += ("Power exported: [power_drained] kilowatts<br>")
+	dat += ("Points earned from exports: [power_drained/200] points<br>")
+	switch(drain_rate)
+		if(0 to 200)
+			rating = "TERRIBLE"
+		if(201 to 400)
+			rating = "BAD"
+		if(401 to 800)
+			rating = "SUBPAR"
+		if(801 to 2500)
+			rating = "DECENT"
+		if(2501 to 6000)
+			rating = "ROBUST"
+		if(6001 to 10000)
+			rating = "THE 1%"
+		if(10001 to 9999999)
+			rating = "UNFATHOMABLE!"

```

Comment:
Personally, I'd have gone with inconceivable, but that's just because I like The Princess Bride.